src/Microsoft.ComponentDetection.Detectors/pip/Contracts/PipDependencySpecification.cs

```diff
@@ -197,6 +197,33 @@ public bool PackageConditionsMet(Dictionary<string, string> pythonEnvironmentVar
         return conditionsMet;
     }
 
+    /// <summary>
+    /// Iterates through the package versions that are explicitly stated, and returns
+    /// the highest version that adheres to the version requirements.
+    /// </summary>
+    /// <example>
+    /// DependencySpecifiers: (&gt;=1.2.3, !=1.2.4, &lt;2.0.0)
```

Comment:
Thoughts on whether this should just detect for `==` and `===`, instead of grabbing package versions that match with `>=` and `<=` as well?

Reply:
Mind elaborating a bit more on the difference here? I would expect it the package version range to be more precise than just detecting explicit references, but I may be missing more context here.

Reply:
Sure, so if the requirements have an explicit `==` or `===`, then we know that the pip installation would pull that package. If it has `>=` by contrast, it is likely that the actual pip installation will pull the latest version from the feed, which may not be the version stated in the requirements file. For example:

take `>=1.2.3, !=1.2.4, <2.0.0` from this doc. We can't reach out to the feed itself to determine any versions, so we only have these 3 versions to go from. The only matching one is `1.2.3`, BUT the feed may have a `1.3.0` for all we know. 

Basically I think it boils down to whether we want to include versions that _may_ be resolved by a pip install, or only the ones we _know_ will be.

Reply:
Ah I follow now. I think it makes sense to follow the matching version, which would be 1.2.3 in your example. 

If customer wants to be explicit we can drop verbose logs that package was added due to naive detection strategy and explicit version range consumption policy. If they truly aren't needing it anymore, they can update the manifest accordingly.